File: app-components/src/externalTesting/java/org/synsystems/onlypass/components/EnvironmentModule.java
package org.synsystems.onlypass.components;

import dagger.Module;
import dagger.Provides;

@Module
@AppScope
public class EnvironmentModule {
  @Provides
  @AppScope
  public Environment provideEnvironment() {
    return Environment.EXTERNAL_TESTING;
  }
}

File: app-components/src/main/java/org/synsystems/onlypass/components/EnvironmentModule.kt
package org.synsystems.onlypass.components

import dagger.Module
import dagger.Provides
import org.synsystems.onlypass.components.Environment.DEV
import org.synsystems.onlypass.components.Environment.EXTERNAL_TESTING
import org.synsystems.onlypass.components.Environment.INTERNAL_TESTING
import org.synsystems.onlypass.components.Environment.PROD

@Module
class EnvironmentModule {

  @Provides
  @AppScope
  fun provideEnvironment(): Environment {
    return when (BuildConfig.BUILD_TYPE) {
      "dev" -> DEV
      "internalTesting" -> INTERNAL_TESTING
      "externalTesting" -> EXTERNAL_TESTING
      "prod" -> PROD
      else -> throw RuntimeException("Unhandled build type ${BuildConfig.BUILD_TYPE}.")
    }
  }
}